geode-core/src/integrationTest/java/org/apache/geode/cache/query/internal/index/IndexMaintenanceJUnitTest.java

```diff
@@ -1057,6 +1071,394 @@ public void testBug43597() throws Exception {
+  @Test
+  public void testIndexClearanceOnMapClear() {
+    try {
+      CacheUtils.getCache();
+      isInitDone = false;
+      Query q =
+          qs.newQuery("SELECT DISTINCT * FROM " + SEPARATOR + "portfolio where status = 'active'");
+      QueryObserverHolder.setInstance(new QueryObserverAdapter() {
+
+        @Override
+        public void afterIndexLookup(Collection coll) {
+          IndexMaintenanceJUnitTest.this.indexUsed = true;
+        }
+      });
+      SelectResults set = (SelectResults) q.execute();
+      if (set.size() == 0 || !this.indexUsed) {
+        fail("Either Size of the result set is zero or Index is not used ");
+      }
+      this.indexUsed = false;
+
+      region.clear();
+      set = (SelectResults) q.execute();
+      if (set.size() != 0 || !this.indexUsed) {
+        fail("Either Size of the result set is not zero or Index is not used ");
+      }
+    } catch (Exception e) {
+      e.printStackTrace();
+      fail(e.toString());
+    } finally {
+      isInitDone = false;
+      CacheUtils.restartCache();
+    }
+  }
+
+  // Asif : Tests for Region clear operations on Index in a Local VM for cases
+  // when a clear
+  // operation & region put operation occur concurrentlty
+  @Test
+  public void testConcurrentMapClearAndRegionPutOperation() {
+    try {
+      CacheUtils.getCache();
+      isInitDone = false;
+      Query q =
+          qs.newQuery("SELECT DISTINCT * FROM " + SEPARATOR + "portfolio where status = 'active'");
+      QueryObserverHolder.setInstance(new QueryObserverAdapter() {
+
+        @Override
+        public void afterIndexLookup(Collection coll) {
+          IndexMaintenanceJUnitTest.this.indexUsed = true;
+        }
+
+        @Override
+        public void beforeRerunningIndexCreationQuery() {
+          // Spawn a separate thread here which does a put opertion on region
+          Thread th = new Thread(new Runnable() {
+
+            @Override
+            public void run() {
+              // Assert that the size of region is now 0
+              assertTrue(region.size() == 0);
+              region.put("" + 8, new Portfolio(8));
+            }
+          });
+          th.start();
+          ThreadUtils.join(th, 30 * 1000);
+          assertTrue(region.size() == 1);
+        }
+      });
+      SelectResults set = (SelectResults) q.execute();
+      if (set.size() == 0 || !this.indexUsed) {
+        fail("Either Size of the result set is zero or Index is not used ");
+      }
+      this.indexUsed = false;
+      region.clear();
+      set = (SelectResults) q.execute();
+      if (set.size() != 1 || !this.indexUsed) {
+        fail("Either Size of the result set is not one or Index is not used ");
+      }
+    } catch (Exception e) {
+      e.printStackTrace();
+      fail(e.toString());
+    } finally {
+      isInitDone = false;
+      CacheUtils.restartCache();
+    }
+  }
+
+  @Test
+  public void testIndexUpdate() {
```

Comment:
This test doesn't seem to actually be testing anything about the behaviour of indexes, just that adding entries to the region results in more results being returned from a query, so it can probably be safely removed.